File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int smallest = prices[0];
        int profit = 0;
        for(int i=1;i<prices.length;i++){
            profit = Math.max(prices[i]-smallest,profit);
            smallest = Math.min(smallest,prices[i]);
        }
        return profit>0? profit:0;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minProfit=prices[0];
        int maxProfit=0;

        for(int i=1;i<prices.size();i++){
            maxProfit=max(maxProfit,prices[i]-minProfit);
            minProfit=min(minProfit,prices[i]);
        }
        return maxProfit;
    }
};